src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs

```diff
@@ -729,6 +729,56 @@ public PointerTypeSymbol Retarget(PointerTypeSymbol type)
                 return new PointerTypeSymbol(newPointed);
             }
 
+            public FunctionPointerTypeSymbol Retarget(FunctionPointerTypeSymbol type)
+            {
+                var signature = type.Signature;
+                var newReturn = Retarget(signature.ReturnTypeWithAnnotations, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
+                var newRefModifiers = RetargetModifiers(signature.RefCustomModifiers, out bool symbolModified);
+                symbolModified = symbolModified || !signature.ReturnTypeWithAnnotations.IsSameAs(newReturn);
+
+                var newParameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+                ImmutableArray<ImmutableArray<CustomModifier>> newParamModifiers = default;
+
+                var paramCount = signature.ParameterCount;
+                if (paramCount > 0)
+                {
+                    var newParameterTypesBuilder = ArrayBuilder<TypeWithAnnotations>.GetInstance(paramCount);
+                    var newParameterCustomModifiersBuilder = ArrayBuilder<ImmutableArray<CustomModifier>>.GetInstance(paramCount);
+                    bool parametersModified = false;
+
+                    foreach (var parameter in signature.Parameters)
+                    {
+                        var newParameterType = Retarget(parameter.TypeWithAnnotations, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
+                        var newModifiers = RetargetModifiers(parameter.RefCustomModifiers, out bool customModifiersChanged);
+                        newParameterTypesBuilder.Add(newParameterType);
+                        newParameterCustomModifiersBuilder.Add(newModifiers);
+                        parametersModified = parametersModified || !parameter.TypeWithAnnotations.IsSameAs(newParameterType) || customModifiersChanged;
+                    }
+
+                    if (parametersModified)
+                    {
+                        newParameterTypes = newParameterTypesBuilder.ToImmutableAndFree();
+                        newParamModifiers = newParameterCustomModifiersBuilder.ToImmutableAndFree();
+                        symbolModified = true;
+                    }
+                    else
+                    {
+                        newParameterTypesBuilder.Free();
+                        newParameterCustomModifiersBuilder.Free();
+                        newParameterTypes = signature.ParameterTypesWithAnnotations;
+                    }
+                }
+
+                if (symbolModified)
```

Comment:
>symbolModified [](start = 20, length = 14)

It doesn't look like added tests actually verify that this clever way of tracking if anything is changed actually works properly. I believe we are testing when everything is changed and nothing is changed. I think it is worth adding a test when  a change happens in one specific place and go though all of them one by one:
- ref modifier on return
- modifier on return
- return type
- ref modifier on first param (at least two params in total) 
- modifier on first param type (at least two params in total) 
- type on first param (at least two params in total) 
- ref modifier on second param
- modifier on second param type
- type on second param